src/sentry/web/frontend/auth_login.py

```diff
@@ -125,6 +125,11 @@ def handle_sso(self, request):
     @never_cache
     @transaction.atomic
     def handle(self, request):
+        redirect_next = request.GET.get('next', None)
```

Comment:
Take a look at the tests.

There's already validation in `sentry.utils.auth.get_login_redirect`.